subsys/net/lib/aws_fota/src/aws_fota.c

```diff
@@ -417,6 +455,14 @@ int aws_fota_mqtt_evt_handler(struct mqtt_client *const client,
 		if (evt->result != 0) {
 			return 0;
 		}
+		if (evt->param.suback.message_id == SUBSCRIBE_NOTIFY_NEXT) {
+			printk("sub notify_next_topic\n");
+			err = aws_jobs_get_job_execution(client, "$next");
+			if (err) {
+				return err;
+			}
+			return 1;
```

Comment:
What is the reason for not using `-ERRSOMETHING`?

Reply:
Return 1 means we handled the message so there is no need for further processing. 0 means we didn't handle the message further processing is needed.  Couldn't find suitable defines to be honest.

Reply:
IMO it sounds more like 0 should be returned when no further processing is needed, while a non-0 positive return value could indicate that further processing is needed.

Reply:
`EAGAIN`?

Reply:
I'll open another PR with changes to the semantics on how errors are handled